assembly/onpremises-ide-packaging-war-factory/src/main/java/com/codenvy/onpremises/factory/filter/FactoryParamsFilter.java

```diff
@@ -83,8 +97,38 @@ public void doFilter(ServletRequest req, ServletResponse resp, FilterChain filte
             }
 
         } else {
-            dispatchToErrorPage(req, resp, INVALID_FACTORY_URL_PAGE, INVALID_PARAMETER_MESSAGE);
-            return;
+            boolean parameterIsOk = false;
+            // check if there is a matching resolver
+            if (resolvers != null) {
+                Map<String, String> parameters = new HashMap<>();
+                Iterator<Map.Entry<String, String[]>> iterator = params.entrySet().iterator();
+                while (iterator.hasNext()) {
+                    Map.Entry<String, String[]> element = iterator.next();
+                    String key = element.getKey();
+                    String[] array = element.getValue();
+                    // check it's only one item length
+                    if (array.length != 1) {
+                        dispatchToErrorPage(req, resp, INVALID_FACTORY_URL_PAGE,
+                                            String.format(PARAMETRIZED_ILLEGAL_PARAMETER_VALUE_MESSAGE, key, params.get(key)[1]));
+                    }
+                    // add to the single value map
+                    parameters.put(key, array[0]);
+                }
+
+                // now loop on resolvers and see if one is accepting it
+                Optional<FactoryParametersResolver> factoryParametersResolverOptional =
+                        this.resolvers.stream().filter((resolver -> resolver.accept(parameters))).findFirst();
+
+                // matching, it's ok
+                if (factoryParametersResolverOptional.isPresent()) {
```

Comment:
@mshaposhnik in fact it's because there are two cases
if (resolvers != null) {
 then we check we have one
}
if no resolvers then we should throw exception as well

it was to avoid duplicaing the dispatchToErrorPage instruction